docs/tutorial/kubernetes-charm-expressjs.rst

```diff
@@ -0,0 +1,804 @@
+    :start-after: [docs:skopeo-copy]
+    :end-before: [docs:skopeo-copy-end]
+    :dedent: 2
+
+This command contains the following pieces:
+
+- ``--insecure-policy``: adopts a permissive policy that
+  removes the need for a dedicated policy file.
+- ``--dest-tls-verify=false``: disables the need for HTTPS
+  and verify certificates while interacting with the MicroK8s registry.
+- ``oci-archive``: specifies the rock we created for our Express app.
+- ``docker``: specifies the name of the image in the MicroK8s registry.
+
+.. seealso::
+
+    `Ubuntu manpage | skopeo
+    <https://manpages.ubuntu.com/manpages/noble/man1/skopeo.1.html>`_
+
+
+Create the charm
+----------------
+
+From the ``~/expressjs-hello-world`` directory, let's create a new directory
+for the charm and change inside it:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: bash
+    :start-after: [docs:create-charm-dir]
+    :end-before: [docs:create-charm-dir-end]
+    :dedent: 2
+
+Similar to the rock, we'll take advantage of a pre-defined extension in
+Charmcraft with the ``--profile`` flag that caters initial charm files for
+specific web app frameworks. Using the ``expressjs-framework`` profile, Charmcraft
+automates the creation of the files needed for our charm, including a
+``charmcraft.yaml`` project file, ``requirements.txt`` and source code for the
+charm. The source code contains the logic required to operate the Express app.
+
+Initialize a charm named ``expressjs-hello-world``:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: bash
+    :start-after: [docs:charm-init]
+    :end-before: [docs:charm-init-end]
+    :dedent: 2
+
+The files will automatically be created in your working directory.
+
+Check out the contents of ``charmcraft.yaml``:
+
+.. code-block:: bash
+
+    cat charmcraft.yaml
+
+The top of the file should look similar to the following snippet:
+
+.. code-block:: yaml
+    :caption: ~/expressjs-hello-world/charm/charmcraft.yaml
+
+    # This file configures Charmcraft.
+    # See https://juju.is/docs/sdk/charmcraft-config for guidance.
+
+    name: expressjs-hello-world
+
+    type: charm
+
+    base: ubuntu@24.04
+
+    # the platforms this charm should be built on and run on.
+    # you can check your architecture with `dpkg --print-architecture`
+    platforms:
+      amd64:
+      # arm64:
+      # ppc64el:
+      # s390x:
+
+    # (Required)
+    summary: A very short one-line summary of the ExpressJS app.
+
+    ...
+
+Verify that the ``name`` is ``expressjs-hello-world``. Ensure that ``platforms``
+includes the architecture of your host. Edit the ``platforms`` key in the
+project file if required.
+
+Let's pack the charm:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: bash
+    :start-after: [docs:charm-pack]
+    :end-before: [docs:charm-pack-end]
+    :dedent: 2
+
+.. note::
+
+    ``CHARMCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS`` is required while the ExpressJS
+    extension is experimental.
```

Comment:
```suggestion
Let's pack the charm. Because the Express profile is in development, we also need to
enable experimental extensions for Charmcraft:

.. literalinclude:: code/expressjs/task.yaml
    :language: bash
    :start-after: [docs:charm-pack]
    :end-before: [docs:charm-pack-end]
    :dedent: 2
```